go/libkb/util_windows.go

```diff
@@ -81,26 +82,29 @@ func GetDataDir(id GUID, envname string) (string, error) {
 	r0, _, _ := procSHGetKnownFolderPath.Call(uintptr(unsafe.Pointer(&id)), uintptr(0), uintptr(0), uintptr(unsafe.Pointer(&pszPath)))
 	// Sometimes r0 == 0 and there still isn't a valid string returned
 	if r0 != 0 || pszPath == 0 {
-		return "", errors.New("can't get FOLDERIDRoamingAppData")
+		return "", fmt.Errorf("can't get %s", envname)
 	}
 
 	defer coTaskMemFree(pszPath)
 
-	// go vet: "possible misuse of unsafe.Pointer"
-	// Have to cast this Windows string to
-	// a Go array of uint16 here, but we don't yet know the length.
-	rawUnicode := (*[1 << 16]uint16)(unsafe.Pointer(pszPath))[:]
-
-	// utf16.Decode crashes without adjusting the slice length
-	pathLen := 0
-	for i, r := range rawUnicode {
-		if r == 0 {
-			pathLen = i
+	var rawUnicode []uint16
+	for i := uintptr(0); ; i++ {
```

Comment:
Rewrote this since I'm not sure how kosher it is to cast to a pointer to array of size larger than the memory block.

Reply:
This is speculative and doesn't prevent the warning, right?

Reply:
It didn't prevent the warning, it was just to remove the possibility of accessing beyond the string end (e.g., by some compiler optimization that reads multiple uint16s at a time)